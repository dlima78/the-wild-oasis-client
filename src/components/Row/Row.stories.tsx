import type { Meta, StoryObj } from '@storybook/react'

import Row, { RowProps } from '.'

const meta = {
  title: 'Row',
  component: Row,
  argTypes: {
    type: {
      options: ['horizontal', 'vertical'],
      control: { type: 'radio' }
    }
  }
} as Meta<RowProps>

export default meta
type Story = StoryObj<typeof Row>

export const Default: Story = {
  render: () => (
    <Row>
      <button>Botão 1</button>
      <button>Botão 2</button>
    </Row>
  )
}
