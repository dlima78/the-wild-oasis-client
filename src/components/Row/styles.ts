import { css, styled } from 'styled-components'
import { RowProps } from '.'

export type WrapperProps = Pick<RowProps, 'type'>
const WrapperModifiers = {
  horizontal: () => css`
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
  `,
  vertical: () => css`
    flex-direction: column;
    gap: 1.6rem;
  `
}

export const Wrapper = styled.div<WrapperProps>`
  ${({ type }) => css`
    display: flex;

    ${!!type && WrapperModifiers[type]()}
  `}
`
