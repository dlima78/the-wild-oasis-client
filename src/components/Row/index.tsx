import { ReactNode } from 'react'
import * as S from './styles'

export type RowProps = {
  type?: 'horizontal' | 'vertical'
  children: ReactNode | ReactNode[]
}

const Row: React.FC<RowProps> = ({ type, children }: RowProps) => (
  <S.Wrapper data-testid="row" type={type}>
    {children}
  </S.Wrapper>
)

export default Row
