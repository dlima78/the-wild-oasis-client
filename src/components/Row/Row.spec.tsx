import { render, screen } from '@testing-library/react'

import Row from '.'

describe('<Row>', () => {
  it('should render the horizontal Row', () => {
    const { container } = render(
      <Row type="horizontal">
        <button>bot達o 1</button>
        <button>bot達o 2</button>
      </Row>
    )

    expect(screen.getByTestId('row')).toHaveStyleRule('flex-direction', 'row')

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the vertical Row', () => {
    render(
      <Row type="vertical">
        <button>bot達o 1</button>
        <button>bot達o 2</button>
      </Row>
    )

    expect(screen.getByTestId('row')).toHaveStyleRule(
      'flex-direction',
      'column'
    )
  })
})
