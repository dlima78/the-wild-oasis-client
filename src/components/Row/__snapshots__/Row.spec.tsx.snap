// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row> should render the horizontal Row 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

<div
  class="c0"
  data-testid="row"
  type="horizontal"
>
  <button>
    botão 1
  </button>
  <button>
    botão 2
  </button>
</div>
`;
